Update Microsoft.CodeAnalysis.NetAnalyzers version

stephentoub opened the pull request:
- Delete some now unnecessary pragmas
- Enable the new Dictionary.ContainsKey guard rule
- Address one violation of the new rule

nil4 commented on eng/CodeAnalysis.test.globalconfig:
```diff
@@ -792,7 +795,7 @@ dotnet_diagnostic.CA5383.severity = none
 dotnet_diagnostic.CA5384.severity = none
 
-# CA5385: Use Rivest�Shamir�Adleman (RSA) Algorithm With Sufficient Key Size
+# CA5385: Use Rivest–Shamir–Adleman (RSA) Algorithm With Sufficient Key Size
```
```suggestion
# CA5385: Use Rivest-Shamir-Adleman (RSA) Algorithm With Sufficient Key Size
```
This doesn't look right (Unicode em- or en-dash causing encoding havoc?) -- replace with plain dashes.

stephentoub replied:
Thanks:
https://github.com/dotnet/roslyn-analyzers/pull/5985

stephentoub committed:
Update Microsoft.CodeAnalysis.NetAnalyzers version

- Delete some now unnecessary pragmas
- Enable the new Dictionary.ContainsKey guard rule
- Address on violation of the new rule

buyaa-n reviewed (approved):
LGTM, thanks!
